### PR TITLE
feat!: have oldVal passed on notify

### LINKED ISSUE #235: Question: store.listen previous value
Hello,

Any guidance on how to receive the old value on `.listen`? I'm thinking something similar to Vue's watchers, receiving `(newVal, oldVal) => void` which helps compute diff between old/new val, such as directions of a carousel, for example.

Thanks for your help!

### PR DESCRIPTION
**Pull request by gismya**
Adds a new argument to the listener callbacks with the old value of the store.

Continuation of the work from #236. Closes #235

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -113,17 +113,18 @@
...
       },
-      "limit": "804 B"
+      "limit": "818 B"
     }
   ],
   "clean-publish": {
     "cleanDocs": true
-  }
+  },
+  "packageManager": "pnpm@8.15.2+sha256.90bb5d6382cb2cb8b8d4959a076b3953d84d1d94121717097bcd41c71344fa14"
```
Comment (gismya) [inline package.json:129]: I'll remove this before setting it to ready for review, I just don't know a way to set the correct version of pnpm that doesn't create any code in the package.json.

**Comment by gismya** [comment]
@ai - I've expanded on the work of @y-nk and taken the PR to a working, but not fully reviewable state. Just wanted to make sure that the resulting API changes are what you had in mind before I spend the time on polish.

**Comment by ai** [comment]
Looks good.

**Comment by gismya** [comment]
Then I think it's ready for review 😄

**Pull request merged**